### ISSUE TITLE
Cannot reproduce performance reported  in the paper, especially for UCF-101-24

### ISSUE THREAD

**Issue by kinux98** [AKASH2907/pi-consistency-activity-detection#8]
First of all, thanks for sharing this great work!

I have tried to reproduce the performance of UCF-101-24 using your code, but it failed.

(100 epoch, vmap@0.2 :  90.02 vs 95.1 (in the paper) / fmap@0.2 : 84.62 vs 89.1 (in the paper)

In addition, the current version of the code is different from what you described in your README.md, so I'm having trouble reproducing the experiment.

It would be grateful if you could tell me how to run the experiments of UCF-101-24 based on the current version of the code.

**Comment by AKASH2907** [comment]
Explicitly what differences between README and code? Can you please point out the lines so I can look into it?
What is the performance at 0.5 metrics?

**Comment by kinux98** [comment]
1. differences between README and code : 

In the README.md, there are no explicit  training commands for  UCF-101-24 or Jhmdb. ( only with running the code respectively for variance and gradient maps, not for each dataset). 

I hope you to provide more direct and detail instruction for training UCF-101-24 / JHMDB
(for example, python3 main_ucf101.py --args.. ).

Thanks for your help!

**Comment by AKASH2907** [comment]
I see you are running both variance and gradient at the same time.
Can you please run once this command?
```
python main_ucf101.py --epochs 100 --bs 8 --loc_loss dice --lr 1e-4 --pkl_file_label train_annots_20_labeled.pkl --pkl_file_unlabel train_annots_80_unlabeled.pkl --wt_loc 1 --wt_cls 1 --wt_cons 0.1 --const_loss l2 --bv --n_frames 5 --thresh_epoch 11 --exp_id cyclic_variance_maps
```

**Comment by kinux98** [comment]
Hi, i got the following results : 

vmap@0.2/0.5 : JHMDB : 95.94 (94.8) / 65.39 (62.8)    UCF101-24 :   93.18 (95.2) / 56.58 (71.8)
fmap@0.2/0.5  : JHMDB : 86.93 (89.2) / 63.14 (63.6)    UCF101-24 :   88.24 (89.6) / 56.20 (69.8)

(.) indicates result from the paper.

Note that, @0.5 shows seriously low performance than that of @0.2 ...

**Comment by kinux98** [comment]
Ok, I will try to run the experiment again and reopen this issue when i encounter the problems. Thank you for your assistance.

**Comment by AKASH2907** [comment]
Yeah, sorry not able to figure out the issue exactly. You can print all maps once and check. Replace these lines at line 188 in evaluate_ucf101.py file to get scores at all maps. 

```print('IoU f-mAP:')
        #config.write_output('IoU f-mAP:\n')
        for i in range(20):
            print(iou_threshs[i], fmAP[i])
        print(fAP[:, 10])
        print('IoU v-mAP:')
        for i in range(20):
            print(iou_threshs[i], vmAP[i])
        print(vAP[:, 10])
```
